src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -60,20 +83,35 @@ internal sealed class SmartRenameViewModel : INotifyPropertyChanged, IDisposable
         }
     }
 
+    public ICommand GetSuggestionsCommand { get; }
+
     public SmartRenameViewModel(
         IThreadingContext threadingContext,
         IAsynchronousOperationListenerProvider listenerProvider,
 #pragma warning disable CS0618 // Editor team use Obsolete attribute to mark potential changing API
-        ISmartRenameSessionWrapper smartRenameSession)
-#pragma warning restore CS0618
+        ISmartRenameSessionWrapper smartRenameSession,
+#pragma warning restore CS0618,
+        RenameFlyoutViewModel baseViewModel)
     {
         _threadingContext = threadingContext;
+        _listenerProvider = listenerProvider;
         _smartRenameSession = smartRenameSession;
         _smartRenameSession.PropertyChanged += SessionPropertyChanged;
-        var listener = listenerProvider.GetListener(FeatureAttribute.SmartRename);
 
-        var listenerToken = listener.BeginAsyncOperation(nameof(_smartRenameSession.GetSuggestionsAsync));
-        _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+        BaseViewModel = baseViewModel;
+        this.currentIdentifierText = baseViewModel.IdentifierText;
+
+        GetSuggestionsCommand = new DelegateCommand(OnGetSuggestionsCommandExecute, null, threadingContext.JoinableTaskFactory);
+    }
+
+    private void OnGetSuggestionsCommandExecute()
+    {
+        if (_getSuggestionsTask.Status is TaskStatus.RanToCompletion or TaskStatus.Faulted or TaskStatus.Canceled)
```

Comment:
Is this method always invoked by the UI thread? if so could you add `_threadingContext.ThrowIfNotOnUIThread();` here?

Reply:
it is on the UI thread indeed. I will update the PR :)